### PR TITLE
RET-2203

### PR DESCRIPTION
**Pull request by cindychmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-2203
https://tools.hmcts.net/jira/browse/RET-2204

### Change description ###

As an Employment Judge, when I complete the Initial Consideration process, I want a copy of the data to be saved (as a PDF file) so that this is viewable from the Initial Consideration tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/domain/documents/InitialConsiderationData.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.domain.documents;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.experimental.SuperBuilder;
+import java.util.List;
+
+@SuperBuilder
+@Data
+public class InitialConsiderationData {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/domain/documents/InitialConsiderationData.java:10]: JavaDoc

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/domain/documents/InitialConsiderationDocument.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.domain.documents;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+import lombok.experimental.SuperBuilder;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.documents.InitialConsiderationData;
+
+@SuperBuilder
+@Data
+public class InitialConsiderationDocument {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/domain/documents/InitialConsiderationDocument.java:10]: JavaDoc

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                .furtherInfoAnyOtherDirections(caseData.getEtICFurtherInformationHearingAnyOtherDirections())
+                .build();
+
+        InitialConsiderationDocument document = InitialConsiderationDocument.builder()
+                .accessKey(accessKey)
+                .outputName(IC_OUTPUT_NAME)
+                .templateName(IC_SUMMARY_SC_TEMPLATE_NAME)
+                .data(data).build();
+
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java:147]: Duplicate object mapper

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                .icOtherIssuesOrFinalOrders(defaultIfEmpty(caseData.getIcOtherIssuesOrFinalOrders(), null))
+                .build();
+
+        InitialConsiderationDocument document = InitialConsiderationDocument.builder()
+                .accessKey(accessKey)
+                .outputName(IC_OUTPUT_NAME)
+                .templateName(IC_SUMMARY_EW_TEMPLATE_NAME)
+                .data(data).build();
+
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java:176]: Duplicate object mapper

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    private static String getDocumentRequestSC(CaseData caseData, String accessKey) throws JsonProcessingException {
+        InitialConsiderationData data = InitialConsiderationData.builder()
+                .caseNumber(defaultIfEmpty(caseData.getEthosCaseReference(), null))
+                .issuesJurisdiction(defaultIfEmpty(caseData.getEtICJuridictionCodesInvalid(), null))
+                .issuesJurCodesGiveDetails(defaultIfEmpty(caseData.getEtICInvalidDetails(), null))
+                .canProceed(defaultIfEmpty(caseData.getEtICCanProceed(), null))
+                .hearingAlreadyListed(defaultIfEmpty(caseData.getEtICHearingAlreadyListed(), null))
+                .hearingListed(caseData.getEtICHearingListed())
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java:61]: do these also need null checks?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+                .furtherInfoAnyOtherDirections(caseData.getEtICFurtherInformationHearingAnyOtherDirections())
+                .build();
+
+        InitialConsiderationDocument document = InitialConsiderationDocument.builder()
+                .accessKey(accessKey)
+                .outputName(IC_OUTPUT_NAME)
+                .templateName(IC_SUMMARY_SC_TEMPLATE_NAME)
+                .data(data).build();
+
+        OBJECT_MAPPER.setSerializationInclusion(JsonInclude.Include.NON_NULL);
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java:152]: not needed as declared earlier on, may also want to remove the log message further on

**Review by hsjhita — APPROVED** [review]

**Review by robin-ieong — APPROVED** [review]

**Pull request merged**